packages/optimize/ios/src/RCTAEPOptimize.m

```diff
@@ -0,0 +1,242 @@
+  [AEPLog debugWithLabel:TAG message:@"generateDisplayInteractionXdm"];
+  AEPProposition* proposition = [AEPProposition initFromData: dictionary];
+  NSArray<AEPOffer*>* offers = [proposition offers];
+  AEPOffer* offerDisplayed = nil;
+  for(AEPOffer* offer in offers) {
+    if([[offer id] isEqualToString:offerId]){
+      offerDisplayed = offer;
+      break;
+    }
+  }
+  
+  if(offerDisplayed != nil){
+    NSDictionary<NSString*, id>* displayInteractionXdm = [offerDisplayed generateDisplayInteractionXdm];
+    resolve(displayInteractionXdm);
+  } else {
+    reject(@"generateDisplayInteractionXdm", [NSString stringWithFormat:@"Error in generating Display interaction XDM for offer with id: %@", offerId], nil);
+  }
+  
+}
+
+#pragma mark - Helper methods
+
+- (NSArray<AEPDecisionScope*>*) createDecisionScopesArray: (NSArray<NSString*>*) decisionScopes {
+  NSMutableArray<AEPDecisionScope *>* decisionScopesArray = [[NSMutableArray alloc] init];
+  for (NSString* decisionScopeName in decisionScopes) {
+      [decisionScopesArray addObject:[[AEPDecisionScope alloc] initWithName:decisionScopeName]];
+  }
+  return decisionScopesArray;
+}
+
+- (NSDictionary<NSString*, NSDictionary<NSString*, id>*>*) convertPropositionToDict: (AEPProposition*) proposition {
+  NSDictionary<NSString*, id>* propositionDict = [[NSMutableDictionary alloc] init];
+  if(!proposition){
+    return propositionDict;
+  }
+  
+  [propositionDict setValue:proposition.id forKey:@"id"];
+  [propositionDict setValue:proposition.scope forKey:@"scope"];
+  [propositionDict setValue:[proposition scopeDetails] forKey:@"scopeDetails"];
+  
+  
+  NSMutableArray<NSDictionary<NSString*, id>*>* offersArray = [[NSMutableArray alloc] init];
+  for(AEPOffer* offer in proposition.offers){
+    [offersArray addObject:[self convertOfferToDict:offer]];
+  }
+  [propositionDict setValue:offersArray forKey:@"items"];
+  return propositionDict;
+}
+
+- (NSDictionary<NSString*, id>*) convertOfferToDict: (AEPOffer*) offer {
+  NSMutableDictionary<NSString*, id>* offerDict = [[NSMutableDictionary alloc] init];
+  if(!offer){
+    return offerDict;
+  }
+  
+  [offerDict setValue:offer.id forKey:@"id"];
+  if([offer etag] != nil){
+    [offerDict setValue:[offer etag] forKey:@"etag"];
+  }
+  [offerDict setValue:[offer schema] forKey:@"schema"];
+  
+  NSDictionary<NSString*, id>* data = [[NSMutableDictionary alloc] init];
+  [data setValue:[offer id] forKey:@"id"];
+  [data setValue:[self convertOfferTypeToString:[offer type]] forKey:@"format"];
+  [data setValue:[offer content] forKey:@"content"];
+  if([offer language] != nil){
+    [data setValue:[offer language] forKey:@"language"];
+  }
+  if([offer characteristics] != nil){
+    [data setValue:[offer characteristics] forKey:@"characteristics"];
+  }
+  
+  [offerDict setValue:data forKey:@"data"];
+  return offerDict;
+}
+
+- (NSString*) convertOfferTypeToString: (AEPOfferType) offerType {
```

Comment:
Instead of this we should have a method on the enum in native code to convert the value to string.